test/integration/stats_integration_test.cc

```diff
@@ -273,6 +273,8 @@ TEST_P(ClusterMemoryTestRunner, MemoryLargeClusterSizeWithFakeSymbolTable) {
   // 2020/05/13  10531    44425       44600   Refactor resource manager
   // 2020/05/20  11223    44491       44600   Add primary clusters tracking to cluster manager.
   // 2020/06/10  11561    44491       44811   Make upstreams pluggable
+  // 2020/06/23  11559    47707       47771   stats: add histogram for request headers size per
```

Comment:
This is the reason that these memory tests exist. We can now have the conversation about whether the value of having a per-cluster histogram for request sizes is worth consuming 3216 bytes per cluster. Consider that deployments may have 10s of thousands of clusters.

It might have that value but it's not obvious.

Reply:
Great to have such tests tracking memory consumption.

Reply:
Should we add a new entry like `enable_upstream_rq_stats` in [cluster config](https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/cluster/v3/cluster.proto#config-cluster-v3-cluster) ?

Reply:
Made the stat optional.